File: binance4j-vision/src/test/java/com/binance4j/spot/GetTradesTest.java
package com.binance4j.spot;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.List;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import com.binance4j.core.exception.ApiException;
import com.binance4j.core.exception.InvalidDateException;
import com.binance4j.core.exception.NotFoundException;
import com.binance4j.core.market.CandlestickInterval;
import com.binance4j.service.TestService;
import com.binance4j.vision.spot.VisionSpotClient;
import com.binance4j.vision.spot.VisionTrade;

public class GetTradesTest {
	final VisionSpotClient client = TestService.CLIENT;
	final String symbol = TestService.SYMBOL;
	final String year = TestService.YEAR;
	final String month = TestService.MONTH;
	final String day = TestService.DAY;
	CandlestickInterval interval = TestService.INTERVAL;

	@Test
	@DisplayName("It should throw an InvalidDateException")
	void testInvalidDate() {
		Exception exception = assertThrows(ApiException.class,
				() -> client.getTrades(symbol, year, month, "32").getData());

		assertTrue(exception.getMessage().contains(new InvalidDateException().getMessage()));
	}

	@Test
	@DisplayName("It should throw a NotFoundException")
	void testNotFoundSync() {
		Exception exception = assertThrows(ApiException.class,
				() -> client.getTrades(symbol, "1995", month, day).getData());

		assertTrue(exception.getMessage().contains(new NotFoundException().getMessage()));
	}

	@Test
	@DisplayName("It should return the trades")
	void testGetTrades() throws ApiException {
		List<VisionTrade> trades = client.getTrades(symbol, year, month, day).getData();

		assertTrue(trades.size() > 0);
		trades.forEach(c -> {
			assertNotNull(c.getIsBestMatch());
			assertNotNull(c.getIsBuyerMaker());
			assertNotNull(c.getPrice());
			assertNotNull(c.getQuantity());
			assertNotNull(c.getQuoteQuantity());
			assertNotNull(c.getTime());
			assertNotNull(c.getTradeId());
		});
	}
}

File: connectors/src/test/kotlin/com/binance4j/vision/GetTradesTest.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.vision

import com.binance4j.core.exception.ApiException
import com.binance4j.core.exception.InvalidDateException
import com.binance4j.core.exception.NotFoundException
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import kotlin.test.assertNotNull

class GetTradesTest : SpotTest() {
	@Test
	fun testInvalidDate() {
		val exception: Exception = Assertions.assertThrows(
			ApiException::class.java
		) { client.getTrades(symbol, year, month, "32").getData() }
		Assertions.assertTrue(exception.message!!.contains(InvalidDateException().message))
	}
	
	@Test
	fun testNotFoundSync() {
		val exception: Exception = Assertions.assertThrows(
			ApiException::class.java
		) { client.getTrades(symbol, "1995", month, day).getData() }
		Assertions.assertTrue(exception.message!!.contains(NotFoundException().message))
	}
	
	@Test
	@Throws(ApiException::class)
	fun testGetTrades() {
		assertNotNull(client.getTrades(symbol, year, month, day).getData())
	}
}